### PR TITLE
[backend] check user access rights before deletion(#6421)

### PR DESCRIPTION
**Pull request by JeremyCloarec**
### Proposed changes

* We now check that the user has access to all related relations before deleting

### Related issues

* #6421 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by labo-flg** [comment]
This issue happens not only when deleting an object but also when merging 2 entities.

repro case:
* Malware X `targets` City1, relationship marking RED
* Malware X `targets` City2, relationship marking GREEN
* Log in with user only allowed for GREEN, but with capa to merge knowledge
* Merge City1 into City2

--> Merge is ok, but if you look at Malware X victimology, you see a "Restricted" (the City1 that has been deleted by the merge).

**Comment by labo-flg** [comment]
linked to https://github.com/OpenCTI-Platform/opencti/pull/6821

I just make sure we don not crash

**Review by labo-flg — APPROVED** [review]

File: opencti-platform/opencti-graphql/src/database/middleware.js [hunk 1]
```diff
@@ -1318,14 +1318,17 @@ const mergeEntitiesRaw = async (context, user, targetEntity, sourceEntities, tar
 
 const loadMergeEntitiesDependencies = async (context, user, entityIds) => {
   const data = { [INTERNAL_FROM_FIELD]: [], [INTERNAL_TO_FIELD]: [] };
+  const rawDependencies = [];
   for (let entityIndex = 0; entityIndex < entityIds.length; entityIndex += 1) {
     const entityId = entityIds[entityIndex];
     // Internal From
     const listFromCallback = async (elements) => {
-      const findArgs = { toMap: true, baseData: true };
+      const findArgs = { toMap: true };
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/database/middleware.js]: why remove baseData param ? (same question for the other places where baseData has been removed.

Comment (JeremyCloarec) [reply opencti-platform/opencti-graphql/src/database/middleware.js]: unless I missed something during my tests, baseData didn't include markings, which are needed for the userFilterStoreElements in mergeEntities

**Pull request merged**